### ISSUE TITLE
Unable to install mcrypt module with php versions 5.6 or 7.1

### ISSUE THREAD

**Issue by LiamMs** [Homebrew/homebrew-php#4518]
## Description

Unable to install mcrypt module with php versions 5.6 or 7.1

I've attempted the following procedures... 
	brew tap homebrew/dupes
	brew tap homebrew/versions
	brew tap homebrew/homebrew-php
	brew install --with-openssl curl
	brew install --with-homebrew-curl --with-apache php71
	brew install php71-mcrypt php71-imagick

Then followed these instructions from... 
	brew info php71

## Context and Logs

Parameter | Value
--------- | ------------------
**macOS version:**       | _your macOS version (output of `sw_vers -productVersion`)_ 
10.12.6

**Homebrew version:**   | _your Homebrew version (output of `brew --version`)_
Homebrew 1.3.5
Homebrew/homebrew-core (git revision 7d96c; last commit 2017-10-12)

**PHP version in use:** | _your PHP version (output of `/path/to/php -v`)_
**Xcode version:**      | _your Xcode version (output of `xcodebuild -version`)_

**Output of `gcc -v`:**
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.0.0 (clang-900.0.37)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

**Output of `php -v`:**
php -v
PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so' - dlopen(/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so, 9): image not found in Unknown on line 0

Warning: PHP Startup: Unable to load dynamic library '/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so' - dlopen(/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so, 9): image not found in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/opt/php71-imagick/imagick.so' - dlopen(/usr/local/opt/php71-imagick/imagick.so, 9): Library not loaded: /usr/local/opt/imagemagick/lib/libMagickWand-7.Q16HDRI.3.dylib
  Referenced from: /usr/local/opt/php71-imagick/imagick.so
  Reason: image not found in Unknown on line 0

Warning: PHP Startup: Unable to load dynamic library '/usr/local/opt/php71-imagick/imagick.so' - dlopen(/usr/local/opt/php71-imagick/imagick.so, 9): Library not loaded: /usr/local/opt/imagemagick/lib/libMagickWand-7.Q16HDRI.3.dylib
  Referenced from: /usr/local/opt/php71-imagick/imagick.so
  Reason: image not found in Unknown on line 0
PHP Warning:  PHP Startup: mcrypt: Unable to initialize module
Module compiled with build ID=API20160303,NTS
PHP    compiled with build ID=API20160303,TS
These options need to match
 in Unknown on line 0

Warning: PHP Startup: mcrypt: Unable to initialize module
Module compiled with build ID=API20160303,NTS
PHP    compiled with build ID=API20160303,TS
These options need to match
 in Unknown on line 0
PHP 7.1.10 (cli) (built: Oct 12 2017 11:45:23) ( ZTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies

**Build logs:** _Link to public gist created with `brew gist-logs my-formula`_
Error: No available formula with the name "my-formula"

**Comment by LiamMs** [comment]
Thanks for your feedback.  Unfortunately I'm still getting the same error.  I did a fresh installation with your approach as well then php -v where i'm fed the same error messages... 
 
PHP Warning: PHP Startup: Unable to load dynamic library '/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so' - dlopen(/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so, 9): image not found in Unknown on line 0...  etc...

My procedures...
        brew remove --force --ignore-dependencies $(brew list) 
	brew tap homebrew/dupes
	brew tap homebrew/versions
	brew tap homebrew/homebrew-php
	brew install --with-openssl curl
	brew install --with-homebrew-curl --with-apache php71
	brew install --build-from-source php71-mcrypt

**Comment by SMillerDev** [comment]
And `brew install --build-from-source php71-mcrypt` doesn't report any errors?

**Comment by LiamMs** [comment]
I really appreciate your effort to help.  I was preparing for a fresh install of macos (high sierra) and figured with the upgrade I'd a fresh go at my homebrew install as well.  Before doing so, I saw your message and decided to give it another run so I can respond effectively.  I repeated the same procedure above and watched this time more diligently for any errors.

Here's the results... 

Ran: brew install --with-homebrew-curl --with-apache php71
Warning: homebrew/php/php71: --with-apache was deprecated; using --with-httpd instead!

Ran: brew install --build-from-source php71-mcrypt
No warnings or errors... 

Ran :  php -i "(command-line 'phpinfo()')"
or...  php -v
and i still get the same load error message

Thanks again for your help.  I saved the output this time so I can answer further questions.  If you wish to see the entire output and feel that might help I'm happy to post it here.  

Final note: I'm considering following the guide below once I reinstall a fresh copy of the newest OS.  Which will take awhile as I've got a much needed file cleanup project to get back under control.  

https://michaelgracie.com/2016/09/plugging-mcrypt-into-php-on-macos-sierra-10-12/

**Comment by SMillerDev** [comment]
Did you verify the images it tries to load are actually there on the system? `/usr/local/Cellar/php71/7.1.10_21/lib/php/extensions/no-debug-zts-20160303/mcrypt.so` for example

**Comment by SMillerDev** [comment]
Could you try  `brew reinstall --with-homebrew-curl --with-httpd php71`
and then `brew reinstall --build-from-source php71-mcrypt`?

There was a fix added for an issue with httpd just 2 days ago so maybe it helps here

**Comment by LiamMs** [comment]
I managed to get it working.  I'm willing to bet the update you mention was the ultimate resolve.  I'm my case I upgraded to High Sierra then performed the brew install.  By chance anyone else looking for the same solution here's my brew formula...

brew update
brew upgrade
brew tap homebrew/dupes
brew tap josegonzalez/homebrew-php
brew install php71-mcrypt

Make a copy of php.ini.default then remove “default copy” from the end of the file name
Add the following line at the top of php.ini…
extension = /usr/local/Cellar/php71-mcrypt/7.1.10_16/mcrypt.so

May need to verify the above path by searching the /usr, /etc, or /var directories for mcrypt.so 
Mine was found at...  /usr/local/Cellar

Thanks again for all your help.  I wasn't able to try your latest suggestion due to the upgrade.
